addons/hr_expense/models/hr_expense.py

```diff
@@ -181,15 +181,21 @@ def create_expense_from_attachments(self, attachment_ids=None, view_type='tree')
         if any(attachment.res_id or attachment.res_model != 'hr.expense' for attachment in attachments):
             raise UserError(_("Invalid attachments!"))
 
+        product = self.env['product.product'].search([('default_code', '=', 'EXP_GEN')])
+
         for attachment in attachments:
             expense = self.env['hr.expense'].create({
                 'name': attachment.name.split('.')[0],
                 'unit_amount': 0,
+                'product_id': product.id
+            })
+            expense.message_post(body=_('Uploaded Attachment'))
+            attachment.write({
```

Comment:
It is. It's related to the way "register_as_main_attachment()" work. We needed to do this workaround in account_invoice_extract too or the preview won't show correctly. If it is directly linked into the message_post, the main attachment is not set correctly (if I well remember, @lobaudoux can confirm)

Reply:
Yes, I confirm. If you post the attachment in `message_post`, it will be set as the main attachment of the message, not the record itself. This is a problem as the OCR relies on `main_attachment_id` to show the preview and choose which document to send to the server.